### PR TITLE
Tailor CI configuration

### PR DESCRIPTION
**Pull request by omasanori**
- [x] Drop Python 3.6
- [x] Use `actions/checkout@v3`
- [x] Use `actions/setup-python@v4`
- [x] Use bazelisk instead of .deb package

### CODE REVIEW

**Comment by omasanori** [comment]
With https://github.com/beancount/beancount/pull/745 and these changes, CI should become green. ✅

File: .github/workflows/pypi-wheel.yaml [hunk 1]
```diff
@@ -14,13 +14,13 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python: ['3.6', '3.7', '3.8', '3.9', '3.10']
+        python: ['3.7', '3.8', '3.9', '3.10']
```
Comment (trim21) [inline .github/workflows/pypi-wheel.yaml]: need to add 3.11

File: .github/workflows/windows-ci.yaml [hunk 2]
```diff
@@ -7,10 +7,10 @@ jobs:
     runs-on: windows-latest
     strategy:
       matrix:
-        python: ['3.6', '3.7', '3.8', '3.9', '3.10']
+        python: ['3.7', '3.8', '3.9', '3.10']
```
Comment (trim21) [inline .github/workflows/windows-ci.yaml]:
```suggestion
        python: ['3.7', '3.8', '3.9', '3.10', '3.11']
```

File: .github/workflows/linux-ci.yaml [hunk 3]
```diff
@@ -8,10 +8,10 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python: ['3.6', '3.7', '3.8', '3.9', '3.10']
+        python: ['3.7', '3.8', '3.9', '3.10']
```
Comment (trim21) [inline .github/workflows/linux-ci.yaml]:
```suggestion
        python: ['3.7', '3.8', '3.9', '3.10', '3.11']
```

**Pull request merged**